flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActor.java

```diff
@@ -342,10 +342,13 @@ private void sendAsyncResponse(CompletableFuture<?> asyncResponse, String method
 							promise.failure(serializedResult.right());
 						}
 					} else {
-						promise.success(value);
+						promise.success(new Status.Success(value));
```

Comment:
was this silently failing with an exception as well?

Reply:
No it was logged as a failure by Akka.